### PR TITLE
Update material text view

### PR DESCRIPTION
**Pull request by SAnatoliiS**

### CODE REVIEW

**Review by savelichalex — APPROVED** [review]

File: kit/inputs/src/UIMaterialTextView/UIMaterialTextViewFloating.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+import { useAutogrow } from './useAutogrow';
+import { UIMaterialTextViewComment } from './UIMaterialTextViewComment';
+import { useExpandingValue } from './useExpandingValue';
+
+// @inline
+const POSITION_FOLDED: number = 0;
+// @inline
+const POSITION_EXPANDED: number = 1;
+
+// EXPANDED_INPUT_OFFSET = UILayoutConstant.contentOffset - UILayoutConstant.contentInsetVerticalX2
```
Comment (sertony) [inline kit/inputs/src/UIMaterialTextView/UIMaterialTextViewFloating.tsx:24]: What about this comment? Maybe kill it if it's not needed?

File: kit/inputs/src/UIMaterialTextView/useClearButton.tsx [hunk 2]
```diff
@@ -22,9 +25,11 @@ export function useClearButton(
                     onPress={clear}
```
Comment (sertony) [inline kit/inputs/src/UIMaterialTextView/useClearButton.tsx:25]: @SAnatoliiS btw, have you found out why this button wasn't pressing on Android device?

**Review by sertony — APPROVED** [review]
Approved, but we won't merge it until @vetkamak integrates it in Surf ;)

**Comment by savelichalex** [comment]
> Approved, but we won't merge it until @vetkamak integrates it in Surf ;)

I thought it doesn't have breaking changes. Correct me if I'm wrong @SAnatoliiS

**Comment by SAnatoliiS** [comment]
@savelichalex is right. There is no breaking changes, only additional prop - `warning?: boolean`.

File: kit/inputs/src/UIMaterialTextView/UIMaterialTextViewFloating.tsx (hunk 1 above)
Comment (SAnatoliiS) [reply kit/inputs/src/UIMaterialTextView/UIMaterialTextViewFloating.tsx:24]: It says here how this constant is calculated

File: kit/inputs/src/UIMaterialTextView/useClearButton.tsx (hunk 2 above)
Comment (SAnatoliiS) [reply kit/inputs/src/UIMaterialTextView/useClearButton.tsx:25]: Yes, I fixed it.

**Comment by sertony** [comment]
> @savelichalex is right. There is no breaking changes, only additional prop - `warning?: boolean`.

Do we need to apply it anywhere in Surf?
Also about the missing `floating` prop, what component should we use instead when we want to "hide" the label, but show it as the placeholder (as it's done in our current InvitationView in Surf)

**Comment by SAnatoliiS** [comment]
Yes, I forgot about prop `floating`. We need to remove it from surf. Do not mix `Label` and `Placeholder`. If we need to hide Label, we don't need to pass any string to it (I made it optional). `floating={false}` for now is equal to `label=""` or missing `label` prop.

**Comment by savelichalex** [comment]
> Yes, I forgot about prop `floating`. We need to remove it from surf. Do not mix `Label` and `Placeholder`. If we need to hide Label, we don't need to pass any string to it (I made it optional). `floating={false}` for now is equal to `label=""` or missing `label` prop.

woah, I actually like it, that makes a lot of sense! 👍

**Comment by SAnatoliiS** [comment]
I made integration PR in Surf https://github.com/tonlabs/TON-Surf/pull/3201

**Pull request merged**